profitbricksapi/profitbricksapi.php

```diff
@@ -0,0 +1,259 @@
+    const profitBricksCreateSnapshot = "https://api.profitbricks.com/cloudapi/v3/datacenters/[data-center-id]/volumes/[volume-id]/create-snapshot";
+
+    public function __construct()
+    {
+    }
+
+    /**
+     * Reads all information about snapshots from ProfitBricks into usable objects.
+     *
+     * @return Snapshot[]|bool
+     */
+    public function snapshots()
+    {
+        $snapshots = false;
+        $profitBricksSnapshots = $this->readFromProfitBricks(self::profitBricksSnapshotApi);
+        if ($profitBricksSnapshots!==false)
+        {
+            $snapshots = array();
+            foreach ($profitBricksSnapshots as $snapshot)
+            {
+                $snapshots[$snapshot->id] = new Snapshot($snapshot);
+            }
+        }
+        return $snapshots;
+    }
+
+    /**
+     * Reads all information about DataCenters from ProfitBricks into usable objects.
+     *
+     * @return DataCenter[]|bool
+     */
+    public function dataCenters()
+    {
+        $dataCenters = false;
+        $profitBricksDataCenters = $this->readFromProfitBricks(self::profitBricksDataCenterApi);
+        if ($profitBricksDataCenters!==false)
+        {
+            $dataCenters = array();
+            foreach ($profitBricksDataCenters as $dataCenter)
+            {
+                $dataCenters[$dataCenter->id] = new DataCenter($dataCenter);
+            }
+        }
+        return $dataCenters;
+    }
+
+     /**
+     * Reads all information about virtual machines/servers from ProfitBricks into usable objects.
+     *
+     * @param DataCenter $_dataCenter determines the data center for which the virtual machines should be searched.
+     * @return VirtualMachine[]|bool
+     */
+    public function virtualMachines(DataCenter $_dataCenter)
+    {
+        $virtualMachines = false;
+        $api = str_replace("[data-center-id]", $_dataCenter->id, self::profitBricksServerApi);
+        foreach ($this->readFromProfitBricks($api) as $virtualMachine)
+        {
+            $virtualMachines[$virtualMachine->id] = new VirtualMachine($virtualMachine);
+        }
+        return $virtualMachines;
+    }
+
+    /**
+     * Reads all information about virtual disks (called volumes) regarding to a virtual machine from ProfitBricks into usable objects.
+     *
+     * @param string $_dataCenterId determines the data center, in which the virtual machine is.
+     * @param VirtualMachine $_virtualMachine determines the virtual machine, which are the disks attached to.
+     * @return VirtualDisk[]|bool
+     */
+    public function virtualDisks(VirtualMachine $_virtualMachine, $_dataCenterId)
+    {
+        $virtualDisks = false;
+        $api = str_replace("[data-center-id]", $_dataCenterId, self::profitBricksDiskApi);
+        $api = str_replace("[server-id]", $_virtualMachine->id, $api);
+        $latestFullSnapshot = null;
+        foreach ($this->readFromProfitBricks($api) as $virtualDisk)
```

Comment:
Theoretically, it would need here this IF-check for the case that nothing usable gets returned.